implementation/leprechaun/src/internal/controller/healthcheck.dto.ts

```diff
@@ -0,0 +1,17 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export const HealthcheckStatus = {
+  UP: 'UP',
+  DOWN: 'DOWN',
+  UNKNOWN: 'UNKNOWN',
+  OUT_OF_SERVICE: 'OUT_OF_SERVICE',
+} as const;
+
+type Values<Object> = Object[keyof Object];
+
+export class HealthcheckDto {
+  @ApiProperty({ enum: () => Object.keys(HealthcheckStatus), example: HealthcheckStatus.UP })
+  status: Values<typeof HealthcheckStatus>;
+
+  // TODO: add more fields
+}
```

Comment:
Huh? That's somehow worse XD

```suggestion
export type HealthcheckStatus = 'UP' | 'DOWN' | 'UNKNOWN' | 'OUT_OF_SERVICE';

export class HealthcheckDto {
  @ApiProperty({ enum: ['UP', 'DOWN', 'UNKNOWN', 'OUT_OF_SERVICE'], example: 'UP' })
  status: HealthcheckStatus;

  // TODO: add more fields
}
```

Reply:
Huh, that was in one of the examples you gave xD